src/requestBuilder.js

```diff
@@ -250,8 +249,7 @@ var requestBuilder = {
   _getHitsHierarchicalFacetsAttributes: function(state) {
     var out = [];
 
-    return reduce(
-      state.hierarchicalFacets,
+    return state.hierarchicalFacets.reduce(
```

Comment:
Did you make sure that `state.hierarchicalFacets` can never be `undefined` at this point?

Reply:
Yes, none of the members of state can be unset unless manually modified without using the functions